lift/src/main/scala/com/linkedin/lift/lib/PositionBiasUtils.scala

```diff
@@ -0,0 +1,175 @@
+package com.linkedin.lift.lib
+
+import com.linkedin.lift.types.ScoreWithLabelAndPosition
+import org.apache.spark.mllib.stat.KernelDensity
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.expressions.Window
+import org.apache.spark.sql.functions.{broadcast, lit, min, rand}
+import org.apache.spark.sql.{Dataset, SparkSession}
+
+object PositionBiasUtils {
+
+  case class PositionBias(position: Int, positionBias: Double)
+
+  /**
+    * Bandwidth computation based on Silverman's rule for kernel density estimation
+    *
+    * @param data i.i.d. samples
+    * @return bandwidth
+    */
+  def getBandwidth(data: RDD[Double]): Double = {
+    val numSamples = data.count
+    val mean = data.sum / numSamples
+    val stdDev = Math.sqrt((data.map(t => t * t).sum / numSamples) - mean * mean)
+    1.06 * stdDev * math.pow(numSamples, -0.2)
+  }
```

Comment:
If the input here is a `Dataset[Double]` or a `DataFrame`, you can use one of the population or sample stddev estimate that Spark provides.
So you may be able to do something like `df.agg(count($"value").as("numSamples"), stddev_pop($"value").as("stdDev")).select(1.06 * $"stdDev" * pow($"numSamples", -0.2)).head`

Reply:
thanks